eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/strategy/mutable/HashBagWithHashingStrategy.java

```diff
@@ -90,6 +91,20 @@ public static <E> HashBagWithHashingStrategy<E> newBagWith(HashingStrategy<? sup
         return result;
     }
 
+    @Override
+    protected MutableObjectIntMap<T> items()
+    {
+        return this.items;
+    }
+
+    /**
+     * Rehashes every element in the set into a new backing table of the smallest possible size and eliminating removed sentinels.
+     */
+    public void trimToSize()
+    {
+        this.items.compact();
```

Comment:
we also need to use `trimToSize()` once https://github.com/eclipse/eclipse-collections/pull/1403 is merged